pyvista/core/filters/data_set.py

```diff
@@ -5535,6 +5535,56 @@ def separate_cells(self):
         """
         return self.shrink(1.0)
 
+    def extract_cells_by_type(self, cell_types=True, progress_bar=False):
+        """Extract cells of a specified type.
+
+        Given an input vtkDataSet and a list of cell types, produce an output
+        dataset containing only cells of the specified type(s). Note that if the
+        input dataset is homogeneous (e.g., all cells are of the same type) and
+        the cell type is one of the cells specified, then the input dataset is
+        shallow copied to the output.
+
+        The type of output dataset is always the same as the input type. Since
+        structured types of data (i.e., vtkImageData, vtkStructuredGrid,
+        vtkRectilnearGrid, vtkUniformGrid) are all composed of a cell of the
+        same type, the output is either empty, or a shallow copy of the input.
+        Unstructured data (vtkUnstructuredGrid, vtkPolyData) input may produce a
+        subset of the input data (depending on the selected cell types).
+
+        Note this filter can be used in a pipeline with composite datasets to
+        extract blocks of (a) particular cell type(s).
+
+        .. warning::
+            Unlike the filter vtkExtractCells which always produces
+            vtkUnstructuredGrid output, this filter produces the same output
+            type as input type (i.e., it is a vtkDataSetAlgorithm). Also,
+            vtkExtractCells extracts cells based on their ids.
+
+        Parameters
+        ----------
+        cell_types : bool | int | list[int]
```

Comment:
What is the use case for `cell_types=True` here?  Won't this always return the same data object back if it has cells?  My reading of `vtkExtractCellsByType` is that `AddAllCellTypes` could be useful in conjunction with `RemoveCellType` to select all types but one, for example.  If this is correct, I'd suggest dropping this usage from this method.  Instead, we could use a convenience method for generating a list of all cell types using `pyvista.CellType`.  For example, can we do something like:

```py
cell_types = pv.CellType.all_cells(exclude=pv.CellType.QUADRATIC_EDGE)
mesh.extract_cells_by_type(cell_types)
```

Bonus points if the design in `CellType` could allow us to distinguish between different flavors based on some criteria, e.g. dimensionality  `pv.CellType.all_edge_cells().  This is out of scope for this PR, but I'm trying to provide some argument for the design of this method to stay out of the game of trying to configure cell types via input.

Reply:
Resolved this in [220f426](https://github.com/pyvista/pyvista/pull/4204/commits/220f4266f267a89d2c6e0b1794fb1f209f7ab598).